WalletWasabi/Services/WalletService.cs

```diff
@@ -253,6 +253,22 @@ private async void IndexDownloader_NewFilterAsync(object sender, FilterModel fil
 			}
 		}
 
+		public static void ResyncWallet(string walletName, string dataDir, Network network)
+		{
+			string BlocksFolderPath = Path.Combine(dataDir, "Blocks", network.ToString());
+			string TransactionsFolderPath = Path.Combine(dataDir, "Transactions", network.ToString());
+			if (Directory.Exists(BlocksFolderPath))
+			{
+				Directory.Delete(BlocksFolderPath, true);
```

Comment:
Do not delete the blocks, they could be hundreds and shared across multiple wallets. Also, take into account that in case the block file is corrupted for some reason it is re-downloaded.

Reply:
Yes.
This is not just bandwidth intensive, it *might* also reduce network level privacy, when only these blocks are downloaded in the same time window. [The bandwidth issue is enough reason to not delete and re-download blocks]